src/uniprotkb/components/landing-page/StatisticsChart.tsx

```diff
@@ -0,0 +1,230 @@
+
+import {
+  StatisticsItem,
+  StatisticsPayload,
+} from '../statistics-page/StatisticsPage';
+import { LocationToPath, Location } from '../../../app/config/urls';
+import { stringifyUrl } from '../../../shared/utils/url';
+
+const renderPieChart = (
+  svgElement: SVGSVGElement | null,
+  data: StatisticsItem[],
+  history: History
+): void => {
+  // Specify the chart’s dimensions.
+  const width = 400;
+  const height = 300;
+  const margin = 40;
+  let current: d3.PieArcDatum<StatisticsItem>;
+
+  const radius = Math.min(width, height) / 2 - margin;
+
+  // Create the color scale.
+  const color = d3
+    .scaleOrdinal<string, string>()
+    .domain(data.map((d) => d.name))
+    .range([...d3.schemeBlues[5]].reverse());
+
+  // Create the SVG container.
+  const svg = d3
+    .select(svgElement)
+    .attr('width', width)
+    .attr('height', height)
+    .append('g')
+    .attr('transform', `translate(${width / 2},${height / 2})`);
+
+  const g = svg.append('g');
+
+  g.append('g').attr('class', 'slices');
+  g.append('g').attr('class', 'labels');
+  g.append('g').attr('class', 'lines');
+
+  // Create the pie layout and arc generator.
+  const pie = d3
+    .pie<StatisticsItem>()
+    .sort(null)
+    .value((d) => d.count);
+
+  const arc = d3
+    .arc<d3.PieArcDatum<StatisticsItem>>()
+    .outerRadius(radius * 0.8)
+    .innerRadius(0);
+
+  const outerArc = d3
+    .arc<d3.PieArcDatum<StatisticsItem>>()
+    .innerRadius(radius * 0.9)
+    .outerRadius(radius * 0.9);
+
+  // Below is used as a reference only for calculating the polyline start position
+  const midArc = d3
+    .arc<d3.PieArcDatum<StatisticsItem>>()
+    .outerRadius(radius * 0.8)
+    .innerRadius(radius * 0.4);
+
+  const key = (d: d3.PieArcDatum<StatisticsItem>) => d.data.name;
+
+  /* ------- PIE SLICES -------*/
+  const pieData = pie(data);
+  const slice = g
+    .select('.slices')
+    .selectAll<SVGPathElement, d3.PieArcDatum<StatisticsItem>>('path.slice')
+    .data(pieData, key);
+
+  slice
+    .enter()
+    .insert('path')
+    .attr('class', 'slice')
+    .merge(slice)
+    .style('fill', (d) => color(d.data.name))
+    .style('stroke-width', '2px')
+    .transition()
+    .duration(1000)
+    .attrTween('d', (d) => {
+      const interpolate = d3.interpolate(current, d);
+      current = interpolate(0);
+      return (t: number) =>
+        arc(interpolate(t)) !== null ? `${arc(interpolate(t))}` : '';
+    });
+
+  slice.exit().remove();
+
+  /* ------- TEXT LABELS -------*/
+
+  const text = g
+    .select('.labels')
+    .selectAll<SVGTextElement, d3.PieArcDatum<StatisticsItem>>('text')
+    .data(pie(data));
+
+  const midAngle = (d: d3.PieArcDatum<StatisticsItem>) =>
+    d.startAngle + (d.endAngle - d.startAngle) / 2;
+
+  text
+    .enter()
+    .append('text')
+    .attr('dy', '.35em')
+    .style('pointer-events', 'auto')
+    .style('cursor', 'pointer')
+    .style('fill', '#014371')
+    .style('font-weight', 'bold')
```

Comment:
What do you think about setting as much as possible the static things in CSS instead of programatically in javascript? It removes a bit of logic, not sure if that would make such a difference perfomance-wise, but at least there's less stuff in the javascript part so it might be easier to follow and maintain the important logic.
See src/tools/align/components/results/NewickTree.tsx & src/tools/align/components/results/styles/NewickTree.scss as an example